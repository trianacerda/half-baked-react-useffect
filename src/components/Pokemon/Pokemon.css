figure {
  display: flex;
  flex-direction: column;
  margin: 10px;
  justify-content: center;
  align-items: center;
  background-color: #2c061f;
}

figure img {
  height: 400px;
  object-fit: contain;
  padding: 5px;
}

.details {
  background-color: #374045;
  margin: 0;
  padding: 0;
  width: 100%;
  text-align: center;
  padding: 10px;
  color: #e1d89f;
}
