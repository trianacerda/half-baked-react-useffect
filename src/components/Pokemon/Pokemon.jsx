import PropTypes from 'prop-types'
import './Pokemon.css'

export default function Pokemon({ pokemon }) {
  const { name, abilityOne, abilityTwo, typeOne, typeTwo, pic, generation } = pokemon

  const hasSecondType = typeTwo !== 'NA'
  const hasSecondAbility = abilityTwo !== 'NA'

  return (
    <figure aria-label="pokemon">
      <img src={pic} alt={`${name}-${generation}`} />
      <div className="details">
        <h2>{name}</h2>
        <p>{generation}</p>
        <p>
          {abilityOne}
          {hasSecondAbility && `/${abilityTwo}`}
        </p>
        <p>
          {typeOne}
          {hasSecondType && `/${typeTwo}`}
        </p>
      </div>
    </figure>
  )
}

Pokemon.propTypes = {
  pokemon: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    abilityOne: PropTypes.string.isRequired,
    abilityTwo: PropTypes.string.isRequired,
    typeOne: PropTypes.string.isRequired,
    typeTwo: PropTypes.string.isRequired,
    pic: PropTypes.string.isRequired,
    generation: PropTypes.string.isRequired,
  }).isRequired,
}
