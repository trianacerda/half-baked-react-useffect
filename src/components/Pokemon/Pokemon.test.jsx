import { screen, render } from '@testing-library/react'
import Pokemon from './Pokemon'

it('Should render a pokemon', () => {
  render(
    <Pokemon
      pokemon={{
        id: 7,
        name: 'Charizard',
        abilityOne: 'blaze',
        abilityTwo: 'NA',
        typeOne: 'fire',
        typeTwo: 'flying',
        pic: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/006.png',
        generation: 'First Generation',
      }}
    />
  )

  const pokemon = screen.getByText(/Charizard/)
  expect(pokemon).toMatchSnapshot()
})
