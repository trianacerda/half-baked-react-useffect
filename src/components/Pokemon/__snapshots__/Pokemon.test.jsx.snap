// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a pokemon 1`] = `
<h2>
  Charizard
</h2>
`;
